ui/text_displayer.js

```diff
@@ -361,4 +365,75 @@ shaka.ui.TextDisplayer = class {
       panelStyle.height = cue.size + '%';
     }
   }
+
+  /**
+   * Returns info about provided lengthValue
+   * @example 100px => { value: 100, unit: 'px' }
+   * @param {?string} lengthValue
+   *
+   * @return {?{ value: number, unit: string }}
+   * @private
+   */
+  static getLengthValueInfo_(lengthValue) {
+    const matches = new RegExp(/(\d*\.?\d+)([a-z]+|%+)/).exec(lengthValue);
+
+    if (!matches ) {
+      return null;
+    }
+
+    return {
+      value: Number(matches[1]),
+      unit: matches[2],
+    };
+  }
+
+  /**
+   * Converts length value
+   *
+   * @param {string} lengthValue
+   * @param {!shaka.extern.Cue} cue
+   * @param {HTMLElement} videoContainer
+   * @return {string}
+   * @private
+  */
+  static convertLengthValue_(lengthValue, cue, videoContainer) {
+    const {TextDisplayer} = shaka.ui;
+
+    const lengthValueInfo = TextDisplayer.getLengthValueInfo_(lengthValue);
+
+    if (!lengthValueInfo) {
+      return lengthValue;
+    }
+
+    const {unit, value} = lengthValueInfo;
+
+    switch (unit) {
+      case '%':
+        return TextDisplayer.getComputedValue_(
+            value / 100,
+            cue,
+            videoContainer
+        );
+      case 'c':
+        return TextDisplayer.getComputedValue_(value, cue, videoContainer);
+      default:
+        return lengthValue;
+    }
+  }
+
+  /**
+   * Returns computed value
+   * @param {number} value
+   * @param {!shaka.extern.Cue} cue
+   * @param {HTMLElement} videoContainer
+   * @return {string}
+   *
+   * @private
+   * */
+  static getComputedValue_(value, cue, videoContainer) {
```

Comment:
"computed value" is a bit vague.  This method actually does something very specific: computes an absolute value in pixels based on cell and video size.

Reply:
was renamed to `getAbsoluteLengthInPixels_` and added the description for this method,
please take a look if I need to add more details
thanks